<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Bootstrap Test for Independence Testing in High Dimensions • hdIndep</title>
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="deps/headroom-0.11.0/headroom.min.js"></script><script src="deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="deps/search-1.0.0/fuse.min.js"></script><script src="deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="Bootstrap Test for Independence Testing in High Dimensions">
<meta name="description" content="This R package implements the block multiplier bootstrap test for pairwise independence of one random variable from a large pool of other random variables. Moreover, this test is combined with a stepdown procedure (Romano-Wolf) for selecting those variables from the pool that violate independence, while controlling for the familiwise error rate. The theory is is developed in Olivares, Olma, and Wilhelm (2025) &lt;https://arxiv.org/pdf/2503.21715#page=1.38&gt;.">
<meta property="og:description" content="This R package implements the block multiplier bootstrap test for pairwise independence of one random variable from a large pool of other random variables. Moreover, this test is combined with a stepdown procedure (Romano-Wolf) for selecting those variables from the pool that violate independence, while controlling for the familiwise error rate. The theory is is developed in Olivares, Olma, and Wilhelm (2025) &lt;https://arxiv.org/pdf/2503.21715#page=1.38&gt;.">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="index.html">hdIndep</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="articles/hdIndep.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/mauolivares/hdIndep/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header"><h1 id="hdindep-a-powerful-bootstrap-test-of-independence-in-high-dimensions">hdIndep: A Powerful Bootstrap Test of Independence in High Dimensions<a class="anchor" aria-label="anchor" href="#hdindep-a-powerful-bootstrap-test-of-independence-in-high-dimensions"></a>
</h1></div>
<div class="section level2">
<h2 id="setup">Setup<a class="anchor" aria-label="anchor" href="#setup"></a>
</h2>
<ol style="list-style-type: decimal">
<li>Clone the repo</li>
<li>Open it in RStudio</li>
<li>On the right panel there is a new tab “Build” (next to “Git”). Click on it</li>
<li>Click Install button</li>
<li>(Optional): More -&gt; Configure Build Tools -&gt; Generate documentation with Roxygen</li>
</ol>
<p>Alternatively:</p>
<ol style="list-style-type: decimal">
<li>Clone the repo</li>
<li>Execute in R console:</li>
</ol>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">install.packages</span>(<span class="sc">&lt;</span>path<span class="sc">-</span>to<span class="sc">-</span>cloned<span class="sc">-</span>repository<span class="sc">&gt;</span>, <span class="at">repos=</span><span class="cn">NULL</span>)</span></code></pre></div>
<p>Alternatively:</p>
<ol style="list-style-type: decimal">
<li>Execute in R console:</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://devtools.r-lib.org/" class="external-link">devtools</a></span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"https://github.com/mauolivares/hdIndep.git"</span><span class="op">)</span></span></code></pre></div>
<p>After successful installation, you can use the functions from <code>hdIndep</code> repository as with a usual package installed from CRAN:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/mauolivares/hdIndep" class="external-link">hdIndep</a></span><span class="op">)</span></span>
<span><span class="op">?</span><span class="va">stepdown_RomanoWolf</span> <span class="co"># Read the documentation</span></span>
<span><span class="fu"><a href="reference/stepdown_RomanoWolf.html">stepdown_RomanoWolf</a></span><span class="op">(</span><span class="va">...</span><span class="op">)</span></span></code></pre></div>
<p>You can also run a tiny simulation in your terminal (~10 seconds) to verify that all is up and running. Simply navigate to the base directory, i.e., set the working directory to the directory containing this README, and execute</p>
<pre><code>cd scripts
Rscript tiny_simulation.r</code></pre>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>The main functions to be used in empirical applications are <code>stepdown_RomanoWolf.r</code> and <code>BMB.cv.r</code>. Their source files are in <code>R/</code> directory. Their documentation could be read from source files or, after installing the <code>hdIndep</code> package, in R by typing, for instance, <code><a href="reference/stepdown_RomanoWolf.html">?stepdown_RomanoWolf</a></code>, like with usual functions.</p>
<p>The example usage of whole workflow is in <code>scripts/tiny_simulation.r</code>.</p>
</div>
<div class="section level2">
<h2 id="data">Data<a class="anchor" aria-label="anchor" href="#data"></a>
</h2>
<p>The vignette <code>vignettes/hdIndep.Rmd</code> presents a step-by-step guide to our proposed stepdown procedure. We use data from two sources:</p>
<ol style="list-style-type: decimal">
<li><p>Hughes et al. (2009) collected liver tissue samples from mice at hourly intervals over a 48-hour period, pooling samples from 3-5 mice at each point. These data are available from Gene Expression Omnibus (GEO). In this paper, we focus on liver dataset (accession GSE11923). We extracted the data using the GEOquery and BiocManager R Packages.</p></li>
<li><p>When comparing the transcripts identified by our method against those from Hughes et al. (2009), we use the transcripts as reported by the authors in the original study. This data set is available from Hughes et al. (2009); see Table S1 in Associated Data section of their paper. The data set was downloaded from the link <a href="https://pmc.ncbi.nlm.nih.gov/articles/PMC2654964/#s2" class="external-link uri">https://pmc.ncbi.nlm.nih.gov/articles/PMC2654964/#s2</a> . The final data set is available in <code>data/pgen.1000442.s012.xls</code>.</p></li>
</ol>
<p>Data Citation:</p>
<p>Hughes, M. E., DiTacchio, L., Hayes, K. R., Vollmers, C., Pulivarthy, S., Baggs, J. E., Satchidananda, P. &amp; Hogenesch, J. B. (2009). Harmonics of circadian gene transcription in mammals. PLoS genetics, 5(4), e1000442.</p>
</div>
</div>
  </main><aside class="col-md-3"><div class="links">
<h2 data-toc-skip>Links</h2>
<ul class="list-unstyled">
<li><a href="https://github.com/mauolivares/hdIndep/" class="external-link">Browse source code</a></li>
<li><a href="https://github.com/mauolivares/hdIndep/issues" class="external-link">Report a bug</a></li>
</ul>
</div>

<div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>
<a href="https://www.r-project.org/Licenses/GPL-3" class="external-link">GPL-3</a> | file <a href="LICENSE-text.html">LICENSE</a>
</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing hdIndep</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Mauricio Olivares <br><small class="roles"> Author, maintainer </small>  </li>
<li>Tomasz Olma <br><small class="roles"> Author </small>  </li>
<li>Daniel Wilhelm <br><small class="roles"> Author </small>  </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Mauricio Olivares, Tomasz Olma, Daniel Wilhelm.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer>
</div>





  </body>
</html>
